sdk/eventhub/azure-eventhub-checkpointstoretable/azure/eventhub/extensions/checkpointstoretable/_tablestoragecs.py

```diff
@@ -17,22 +56,299 @@ class TableCheckpointStore:
+
+    def _update_ownership(self, ownership, **kwargs):
+        """_update_ownership mutates the passed in ownership."""
+        try:
+            ownership_entity = TableCheckpointStore._create_ownership_entity(ownership)
+            metadata = self._table_client.update_entity(
+                mode=UpdateMode.REPLACE,
+                entity=ownership_entity,
+                etag=ownership["etag"],
+                match_condition=MatchConditions.IfNotModified,
+                **kwargs
+            )
+            ownership["etag"] = metadata["etag"]
+            updated_entity = self._table_client.get_entity(
+                partition_key=ownership_entity["PartitionKey"],
+                row_key=ownership_entity["RowKey"],
+                **kwargs
+            )
+            ownership["last_modified_time"] = _to_timestamp(
+                updated_entity.metadata.get("timestamp")
+            )
+            return ownership
+        except (ResourceNotFoundError, ValueError):
+            metadata = self._table_client.create_entity(
+                entity=ownership_entity, headers={"Prefer": "return-content"}, **kwargs
+            )
+            ownership["etag"] = metadata["etag"]
+            ownership["last_modified_time"] = _to_timestamp(
+                dateutil.parser.isoparse(metadata["content"]["Timestamp"])
+            )
+            return ownership
+
+    def _claim_one_partition(self, ownership, **kwargs):
+        new_ownership = ownership.copy()
+        try:
+            self._update_ownership(new_ownership, **kwargs)
+            return new_ownership
+        except (ResourceModifiedError, ResourceExistsError):
+            logger.info(
+                "EventProcessor instance %r of namespace %r eventhub %r consumer group %r "
+                "lost ownership to partition %r",
+                new_ownership["owner_id"],
+                new_ownership["fully_qualified_namespace"],
+                new_ownership["eventhub_name"],
+                new_ownership["consumer_group"],
+                new_ownership["partition_id"],
+            )
+            raise OwnershipLostError()
+        except Exception as error:  # pylint:disable=broad-except
+            logger.warning(
+                "An exception occurred when EventProcessor instance %r claim_ownership for "
+                "namespace %r eventhub %r consumer group %r partition %r. "
+                "The ownership is now lost. Exception "
+                "is %r",
+                new_ownership["owner_id"],
+                new_ownership["fully_qualified_namespace"],
+                new_ownership["eventhub_name"],
+                new_ownership["consumer_group"],
+                new_ownership["partition_id"],
+                error,
+            )
+            return new_ownership  # Keep the ownership if an unexpected error happens
+
+    def list_ownership(
+        self, fully_qualified_namespace, eventhub_name, consumer_group, **kwargs
+    ):
```

Comment:
need a type hint here:
```suggestion
    ):
    # type: (str, str, str, Any) -> Iterable[Dict[str, Any]]
```